pkg/access_token/access_token_file_test.go

```diff
@@ -19,138 +19,200 @@ func TestAccessTokenFile(t *testing.T) {
+
+						Convey("Finishes without raising an error", func() {
+							So(err, ShouldEqual, nil)
+						})
+
+						Convey("Properly deletes the file", func() {
+							_, err = os.Stat("/tmp/accessTokenFileDel1")
+							So(err, ShouldNotEqual, nil)
+						})
+
+						Convey("Properly clears all data from memory", func() {
+							// Both input & output arrays should be cleared from memory
+							empty := make([]byte, len(dataActual))
+							eq := reflect.DeepEqual(dataActual, empty)
+							So(eq, ShouldEqual, true)
+							eq = reflect.DeepEqual(dataFromRead, empty)
+							So(eq, ShouldEqual, true)
+						})
+					})
+				})
+			})
 		})
 
-		Convey("Returns no error if Data input is nil", func() {
+		Convey("Given an access token with no data saved to a file", func() {
 			tokenInFile.Data = nil
 			os.Create("/tmp/accessTokenFileDel2")
 			tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel2"
-			err := tokenInFile.Delete()
-			So(err, ShouldEqual, nil)
 
-			// Check that file is not exits
-			_, err = os.Stat("/tmp/accessTokenFileDel2")
-			So(err, ShouldNotEqual, nil)
-		})
+			Convey("When running the Delete method", func() {
+				err := tokenInFile.Delete()
 
-		Convey("Returns error if file already deleted", func() {
-			tokenInFile.Data = nil
-			os.Create("/tmp/accessTokenFileDel3")
-			tokenInFile.TokenFilePath = "/tmp/accessTokenFileDel3"
-			err := tokenInFile.Delete()
-			So(err, ShouldEqual, nil)
-
-			// Check that file is not exits
-			err = tokenInFile.Delete()
-			So(err.Error(), ShouldEqual, "error deleting access token")
+				Convey("Deletes the file and no errors are raised", func() {
+					So(err, ShouldEqual, nil)
+
+					// Check that file does not exist
+					_, err = os.Stat("/tmp/accessTokenFileDel2")
+					So(err, ShouldNotEqual, nil)
+				})
+
+				Convey("When running the Delete method again on the same file", func() {
+					err = tokenInFile.Delete()
+
+					Convey("Finishes with an error deleting the access token", func() {
+						So(err.Error(), ShouldEqual, "error deleting access token")
+					})
+				})
+			})
 		})
 
-		Convey("Returns no error if delete data from proxy struct is as expected", func() {
-			// Write Data to source interface
-			dataActual := []byte{'t', 'e', 's', 't'}
-			tokenInFile.Write(dataActual)
+		Convey("Given two instances of the accessTokenHandler interface", func() {
+
+			Convey("Finishes without raising errors and data has been deleted", func() {
+				// Write Data to source interface
+				dataActual := []byte{'t', 'e', 's', 't'}
+				tokenInFile.Write(dataActual)
 
-			// Set proxy struct with source interface
-			var proxyStruct ProxyHandlerTokenFile
-			proxyStruct.AccessToken = tokenInFile
+				// Set proxy struct with source interface
+				var proxyStruct ProxyHandlerTokenFile
+				proxyStruct.AccessToken = tokenInFile
 
-			// Delete access token from proxy
-			err := proxyStruct.AccessToken.Delete()
-			So(err, ShouldEqual, nil)
+				// Delete access token from proxy
```

Comment:
same here. anywhere you have a comment explaining the step is a good point to add a convey